ironfist/src/cpp/scripting/scripting.cpp

```diff
@@ -772,38 +772,116 @@ char *GetScriptContents() {
   return script_contents;
 }
 
-void GetNextMapVariable(int &key, const char *&mapVariableId, const char *&mapVariableValue) {
-  // PUSH THE MAP VARIABLES LIST TO THE TOP OF THE STACK OR nil IF THERE IS NO LIST
-  lua_getglobal(map_lua, "mapVariables");
-
-  // CHECK IF THERE IS INDEED A LIST OF MAP VARIABLES
-  if (lua_isnil(map_lua, -1)) {
-    mapVariableId = "noMapVariables";
-    return;
-  }
-
-  // GO TO THE mapVariable INDEXED BY key
-  if (key >= 0) {
-    lua_pushnumber(map_lua, key);
-  }
-  else {
-    lua_pushnil(map_lua);
-  }
-  if (lua_next(map_lua, -2)) {
-    key = lua_tointeger(map_lua, -2);
-    mapVariableId = lua_tostring(map_lua, -1);
-    lua_pop(map_lua, 2);
-    lua_getglobal(map_lua, mapVariableId);
-    mapVariableValue = lua_tostring(map_lua, -1);
-  }
-  else {
-    mapVariableId = "noMapVariables";
-  }
-}
-
-void SetMapVariables(const char *id, const char *quantity) {
-  lua_pushstring(map_lua, quantity);
-  lua_setglobal(map_lua, id);
+bool isTable(std::string valueType) {

```

Comment:
It makes perfect sense locally to ask if a string is a table, but not generally. Make this function "static" to emphasize this.

Most of the codebase doesn't do this, but I'd like to adopt this convention more going forward. I'm putting a not in the style guidelines.

Also, please have it accept a string& instead of a raw string. The way you have it now, the value will be copied every time you invoke this function.